app/assets/javascripts/govuk_publishing_components/modules.js

```diff
@@ -34,23 +34,19 @@
           var moduleName = camelCaseAndCapitalise(moduleNames[j])
           var started = element.data(moduleNames[j] + '-module-started')
 
-          if ( // GOV.UK Publishing & Legacy Modules
-            typeof GOVUK.Modules[moduleName] === 'function' &&
-            !GOVUK.Modules[moduleName].prototype.init &&
-            !started
-          ) {
-            module = new GOVUK.Modules[moduleName]()
-            module.start(element)
-            element.data(moduleNames[j] + '-module-started', true)
-          }
+          if (typeof GOVUK.Modules[moduleName] === 'function' && !started) {
+            // GOV.UK Legacy Modules using jQuery
+            if (!GOVUK.Modules[moduleName].prototype.init) {
+              module = new GOVUK.Modules[moduleName]()
+              module.start(element)
+              element.data(moduleNames[j] + '-module-started', true)
+            }
 
-          if ( // GOV.UK Frontend Modules
-            typeof GOVUK.Modules[moduleName] === 'function' &&
-            GOVUK.Modules[moduleName].prototype.init &&
-            !started
-          ) {
-            module = new GOVUK.Modules[moduleName](element[0]).init()
-            element.data(moduleNames[j] + '-module-started', true)
+            // Vanilla JavaScript GOV.UK Modules and GOV.UK Frontend Modules
+            if (GOVUK.Modules[moduleName].prototype.init) {
+              module = new GOVUK.Modules[moduleName](element[0]).init()
+              element.data(moduleNames[j] + '-module-started', true)
+            }
```

Comment:
Just a couple of quick suggestions to make this simpler still:

```js    
            if (GOVUK.Modules[moduleName].prototype.init) {
              // Vanilla JavaScript GOV.UK Modules and GOV.UK Frontend Modules
              new GOVUK.Modules[moduleName](element[0]).init()
            } else {
              // GOV.UK Legacy Modules using jQuery
              new GOVUK.Modules[moduleName]().start(element)
            }
            element.data(moduleNames[j] + '-module-started', true)
```

You'd also be able to remove the [`var module` from line 33](https://github.com/alphagov/govuk_publishing_components/blob/72c9be4f7b562783ae68c6b00ef626e5d207d5b5/app/assets/javascripts/govuk_publishing_components/modules.js#L33) since we don't actually do anything with it.

Reply:
I've implemented part of your suggestion, but I'm hesitant about switching from two `if` statements to an `if else` - the original code seemed quite explicit about that. I'm fairly sure there shouldn't be a module that doesn't have either a `start` or an `init` function, but assuming that is something I'd be nervous about.

Reply:
Thanks, there isn't a reason to be hesitant though the code you've got is just an else in disguise. As you have:

`if (!GOVUK.Modules[moduleName].prototype.init) {` and `if (GOVUK.Modules[moduleName].prototype.init) {`

one will definitely run, if one of those conditions resolves to true the other will resolve to false 

The previous code also catered for whether things were started and whether a module existed. Therefore, it was possible for neither of these to run. This is now covered by the conditional on line 36.

Reply:
I checked with @alex-ju and the reason for having two `if` statements is because we now allow more than one module to be declared on an element, and they could be of different types. Having two `if` statements allows them both to run.